docs/hardhat-network/reference/README.md

```diff
@@ -0,0 +1,407 @@
+
+## Initial State
+
+Hardhat Network is initialized by default in this state:
+
+- A brand new blockchain, just with the genesis block.
+- 20 accounts with 10000 ETH each, generated with the mnemonic `"test test test test test test test test test test test junk"`. Their addresses are:
+  - `0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266`
+  - `0x70997970C51812dc3A010C7d01b50e0d17dc79C8`
+  - `0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC`
+  - `0x90F79bf6EB2c4f870365E785982E1f101E93b906`
+  - `0x15d34AAf54267DB7D7c367839AAf71A00a2C6A65`
+  - `0x9965507D1a55bcC2695C58ba16FB37d819B0A4dc`
+  - `0x976EA74026E726554dB657fA54763abd0C3a0aa9`
+  - `0x14dC79964da2C08b23698B3D3cc7Ca32193d9955`
+  - `0x23618e81E3f5cdF7f54C3d65f7FBc0aBf5B21E8f`
+  - `0xa0Ee7A142d267C1f36714E4a8F75612F20a79720`
+  - `0xBcd4042DE499D14e55001CcbB24a551F3b954096`
+  - `0x71bE63f3384f5fb98995898A86B02Fb2426c5788`
+  - `0xFABB0ac9d68B0B445fB7357272Ff202C5651694a`
+  - `0x1CBd3b2770909D4e10f157cABC84C7264073C9Ec`
+  - `0xdF3e18d64BC6A983f673Ab319CCaE4f1a57C7097`
+  - `0xcd3B766CCDd6AE721141F452C550Ca635964ce71`
+  - `0x2546BcD3c84621e976D8185a91A922aE77ECEc30`
+  - `0xbDA5747bFD65F08deb54cb465eB87D40e51B197E`
+  - `0xdD2FD4581271e230360230F9337D5c0430Bf44C0`
+  - `0x8626f6940E2eb28930eFb4CeF49B2d1F2C9C1199`
+
+To customise it, take a look at [the configuration section](/config/README.md#hardhat-network).
+
+## JSON-RPC methods support
+
+### Standard methods
+
+#### `eth_accounts`
+
+#### `eth_blockNumber`
+
+#### `eth_call`
+
+#### `eth_chainId`
+
+#### `eth_coinbase`
+
+#### `eth_estimateGas`
+
+#### `eth_gasPrice`
+
+#### `eth_getBalance`
+
+#### `eth_getBlockByHash`
+
+#### `eth_getBlockByNumber`
+
+#### `eth_getBlockTransactionCountByHash`
+
+#### `eth_getBlockTransactionCountByNumber`
+
+#### `eth_getCode`
+
+#### `eth_getFilterChanges`
+
+#### `eth_getFilterLogs`
+
+#### `eth_getLogs`
+
+#### `eth_getStorageAt`
+
+#### `eth_getTransactionByBlockHashAndIndex`
+
+#### `eth_getTransactionByBlockNumberAndIndex`
+
+#### `eth_getTransactionByHash`
+
+#### `eth_getTransactionCount`
+
+#### `eth_getTransactionReceipt`
+
+#### `eth_mining`
+
+#### `eth_newBlockFilter`
+
+#### `eth_newFilter`
+
+#### `eth_newPendingTransactionFilter`
+
+#### `eth_pendingTransactions`
+
+#### `eth_sendRawTransaction`
+
+#### `eth_sendTransaction`
+
+#### `eth_sign`
+
+#### `eth_signTypedData_v4`
+
+#### `eth_subscribe`
+
+#### `eth_syncing`
+
+#### `eth_uninstallFilter`
+
+#### `eth_unsubscribe`
+
+#### `net_listening`
+
+#### `net_peerCount`
+
+#### `net_version`
+
+#### `web3_clientVersion`
+
+#### `web3_sha3`
+
+### Hardhat network methods
+
+#### `hardhat_addCompilationResult`
+
+Add information about compiled contracts
+
+#### `hardhat_dropTransaction`
+
+Remove a transaction from the mempool
+
+#### `hardhat_getStackTraceFailuresCount`
```

Comment:
I think I wouldn't document this. It's mostly an internal API.

Reply:
Addressed in https://github.com/nomiclabs/hardhat/pull/1657/commits/368a1ff0a7678b23280d5654d9dc67b3fc2d1674